sentry-core/src/client.rs

```diff
@@ -308,11 +322,22 @@ impl Client {
         }
     }
 
+    #[cfg(feature = "UNSTABLE_metrics")]
+    pub(crate) fn send_metric(&self, metric: &str) {
+        if let Some(ref flusher) = *self.metrics_flusher.read().unwrap() {
+            flusher.send_metric(metric)
```

Comment:
I would propose a slight change in the signatures here. Apart from using cadence, we should offer a convenient programmatic interface to capture metrics from code directly. For inspiration, have a look at `sentry_sdk.metrics` in the Python SDK.

1. One basic function to capture metrics with structured inputs: name, unit, value, and tags. This could be a function `capture_metric(&self, metric: Metric)`. See [this](https://github.com/getsentry/relay/blob/e7d779eeebcfe7dd3556bfd37be44428c4815dbb/relay-metrics/src/protocol.rs#L321) type from an older version of Relay for an example. Note how there is a type `MetricValue` separate from `BucketValue` for the user. 
2. Convenience functions for `.count` / `.incr`, `.gauge`, `.time` etc -- either named like in the Sentry SDK or like in cadence.
3. Implement `FromStr for Metric` to support parsing plain statsd payloads (which cannot contain packed values), as required by the cadence sink sadly.
4. The cadence sink could parse into `Metric`, then invoke `capture_metric` on `Hub::current`. This way, the CPU for parsing statsd payload occurs at call site, not in the central background thread which might backlog easier.